pkg/cmd/server/api/validation/oauth.go

```diff
@@ -173,6 +173,9 @@ func ValidateIdentityProvider(identityProvider api.IdentityProvider, fldPath *fi
 		case (*api.GitHubIdentityProvider):
 			validationResults.AddErrors(ValidateOAuthIdentityProvider(provider.ClientID, provider.ClientSecret, identityProvider.UseAsChallenger)...)
 
+		case (*api.GitLabIdentityProvider):
+			validationResults.AddErrors(ValidateOAuthIdentityProvider(provider.ClientID, provider.ClientSecret, identityProvider.UseAsChallenger)...)
```

Comment:
make a `ValidateGitLabIdentityProvider` function that:
1. calls ValidateOAuthIdentityProvider
2. ensures the URL is set and is https
3. ensures the CA file exists if specified